corehq/apps/app_manager/models.py

```diff
@@ -649,25 +649,27 @@ def generate_id(cls):
         return hex(random.getrandbits(160))[2:-1]
 
     @classmethod
-    def get_form(cls, form_unique_id, and_app=False):
-        try:
-            d = get_db().view(
-                'app_manager/xforms_index',
-                key=form_unique_id
-            ).one()
-        except MultipleResultsFound as e:
-            raise XFormIdNotUnique(
-                "xform id '%s' not unique: %s" % (form_unique_id, e)
-            )
-        if d:
-            d = d['value']
-        else:
-            raise ResourceNotFound()
-        # unpack the dict into variables app_id, module_id, form_id
-        app_id, unique_id = [d[key] for key in ('app_id', 'unique_id')]
+    def get_form(cls, form_unique_id, and_app=True):
```

Comment:
Unfortunately you can't cache the form since the pickling destroys some private variables that are used in other places e.g. form._parent which is used in form.get_module()